### PR TITLE
chore: high level api speccing

### PR DESCRIPTION
**Pull request by harrysolovay**

### CODE REVIEW

File: spec/Node.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  StorageMapValue = "StorageMapValue",
+  StorageMapKeysPage = "StorageMapKeysPage",
+  Access = "Access",
+  ExtrinsicFactory = "ExtrinsicFactory",
+  Extrinsic = "Extrinsic",
+  SignedExtrinsic = "SignedExtrinsic",
+  Event = "Event",
+  Read = "Read",
+  Subscribe = "Subscribe",
+}
```
Comment (tjjfvi) [inline spec/Node.ts:17]: I thought I had rid this land of these pests!

File: spec/Extrinsic.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    readonly factory: ExtrinsicFactory<CallData>,
+    readonly caller: MultiAddress,
+    readonly callData: CallData,
+  ) {
+    super();
+  }
+
+  sign(sign: (message: Uint8Array) => Uint8Array): SignedExtrinsic {
+    return new SignedExtrinsic(this as Extrinsic, sign);
+  }
```
Comment (tjjfvi) [inline spec/Extrinsic.ts:21]: Needs to also take `extra` and `additional`

**Comment by tjjfvi** [comment]
I think all of these node classes should be named `FooNode` to reduce clashes

File: spec/Node.ts (hunk 1 above)
Comment (harrysolovay) [reply spec/Node.ts:17]: Whoops. String literals from here on out. Fingers crossed!

**Comment by harrysolovay** [comment]
Closing in favor of #132

**Pull request closed without merging**